TestSizeBoundedForceMerge.testByteSizeLimit test failure

### Description

Came up in continuous testing. It has failed on 9x and main.

It replicates reliably.

```
org.apache.lucene.index.TestSizeBoundedForceMerge > testByteSizeLimit FAILED
    java.lang.AssertionError: expected:<3> but was:<5>
        at __randomizedtesting.SeedInfo.seed([AE91D9F034F31034:2DAA18CA35644BD5]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:647)
        at org.junit.Assert.assertEquals(Assert.java:633)
        at org.apache.lucene.index.TestSizeBoundedForceMerge.testByteSizeLimit(TestSizeBoundedForceMerge.java:83)
```

### Gradle command to reproduce

./gradlew test --tests TestSizeBoundedForceMerge.testByteSizeLimit -Dtests.seed=AE91D9F034F31034 -Dtests.locale=no-NO -Dtests.timezone=America/Martinique -Dtests.asserts=true -Dtests.file.encoding=UTF-8

--- a/lucene/core/src/test/org/apache/lucene/index/TestSizeBoundedForceMerge.java
+++ b/lucene/core/src/test/org/apache/lucene/index/TestSizeBoundedForceMerge.java
@@ -44,7 +44,8 @@ public class TestSizeBoundedForceMerge extends LuceneTestCase {
 
   private static IndexWriterConfig newWriterConfig() {
     IndexWriterConfig conf = newIndexWriterConfig(null);
-    // Force using the default codec and not e.g. SimpleText which has a non-deterministic byte size of its segment info due to escape characters.
+    // Force using the default codec and not e.g. SimpleText which has a non-deterministic byte size
+    // of its segment info due to escape characters.
     conf.setCodec(TestUtil.getDefaultCodec());
     conf.setMaxBufferedDocs(IndexWriterConfig.DISABLE_AUTO_FLUSH);
     conf.setRAMBufferSizeMB(IndexWriterConfig.DEFAULT_RAM_BUFFER_SIZE_MB);

--- a/lucene/core/src/test/org/apache/lucene/index/TestSizeBoundedForceMerge.java
+++ b/lucene/core/src/test/org/apache/lucene/index/TestSizeBoundedForceMerge.java
@@ -23,6 +23,7 @@ import org.apache.lucene.document.StringField;
 import org.apache.lucene.store.ByteBuffersDirectory;
 import org.apache.lucene.store.Directory;
 import org.apache.lucene.tests.util.LuceneTestCase;
+import org.apache.lucene.tests.util.TestUtil;
 
 public class TestSizeBoundedForceMerge extends LuceneTestCase {
 
@@ -43,6 +44,8 @@ public class TestSizeBoundedForceMerge extends LuceneTestCase {
 
   private static IndexWriterConfig newWriterConfig() {
     IndexWriterConfig conf = newIndexWriterConfig(null);
+    // Force using the default codec and not e.g. SimpleText which has a non-deterministic byte size of its segment info due to escape characters.
+    conf.setCodec(TestUtil.getDefaultCodec());
     conf.setMaxBufferedDocs(IndexWriterConfig.DISABLE_AUTO_FLUSH);
     conf.setRAMBufferSizeMB(IndexWriterConfig.DEFAULT_RAM_BUFFER_SIZE_MB);
     // don't use compound files, because the overhead make size checks unreliable.
@@ -67,6 +70,14 @@ public class TestSizeBoundedForceMerge extends LuceneTestCase {
     writer.close();
 
     SegmentInfos sis = SegmentInfos.readLatestCommit(dir);
+    int numberOfSegmentsOfMinimumSize = 1;
+    for (int i = 1; i < sis.size(); ++i) {
+      if (sis.info(i).sizeInBytes() == sis.info(0).sizeInBytes()) {
+        numberOfSegmentsOfMinimumSize++;
+      }
+    }
+    assertEquals(numSegments - 1, numberOfSegmentsOfMinimumSize);
+
     double min = sis.info(0).sizeInBytes();
 
     conf = newWriterConfig();

